### PR TITLE
add vscode settings and extension recommendations

### PR DESCRIPTION
**Pull request by radium-v**
# Pull Request

## 📖 Description

Adds a `.vscode` directory and shared default settings for the project.

### 🎫 Issues

#3136

## 👩‍💻 Reviewer Notes

The settings we add to the file should only be for maintaining formatting consistency and relieving common cross-platform editor misconfigurations.

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

## ⏭ Next Steps

When #4001 is ready to be published, we can add it as a recommendation to the `extensions.json` list.

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "recommendations": [
+    "bierner.github-markdown-preview",
+    "bierner.jsdoc-markdown-highlighting",
+    "dbaeumer.vscode-eslint",
+    "ghmcadams.lintlens",
+    "rvest.vs-code-prettier-eslint",
+    "sidneys1.gitconfig"
```
Comment (EisenbergEffect) [inline .vscode/extensions.json:8]: Should we add extension for `html` and `css`?

**Review by EisenbergEffect — APPROVED** [review]
My only wish is that we could have a tab size of 2 and switch to single quotes 🤣

File: .vscode/extensions.json (hunk 1 above)
Comment (radium-v) [reply .vscode/extensions.json:8]: Let's talk about getting #4001 published, then we can add it here!

**Review by nicholasrice — APPROVED** [review]

**Review by Ibrahimmaga — APPROVED** [review]

File: .vscode/extensions.json (hunk 1 above)
Comment (EisenbergEffect) [reply .vscode/extensions.json:8]: Ok, that is awesome. I don't remember seeing that earlier. Let me take a look at that and see how we can move it forward. I've got another meeting coming up with someone from another team who is interested in working with us on a VS Code plugin. Something that would generate files and default component setup. Maybe we can get this PR in and publish and then use this plugin as a vehicle to deliver further ergonomics for FAST through tools.

**Review by chrisdholt — APPROVED** [review]
@radium-v should we add any documentation on how to ensure these sync on the workspace side of things? I believe this will *prompt* users to install or leverage what's setup for the repo - would documentation help folks understand what to do when prompted and why?

**Comment by radium-v** [comment]
@chrisdholt Would a new page in `docs/integrations` be a good place to add some documentation?

**Comment by chrisdholt** [comment]
> @chrisdholt Would a new page in `docs/integrations` be a good place to add some documentation?

@radium-v I'd probably suggest two potential alternative places:

1. The contributing guide
2. Under "Tools" we have a VSCode page - that one might make the most sense.

**Comment by Ibrahimmaga** [comment]
so we are all good on this one?

**Comment by EisenbergEffect** [comment]
Ping @awentzel for final DevOps review and merge.

**Review by janechu — APPROVED** [review]

**Review by bheston — APPROVED** [review]
I'd like to see these changes go in. I was just wanting to add the spell check including a custom dictionary the other day, but remembered there was this PR to get started.

**Pull request merged**